crates/holochain/src/core/ribosome/host_fn/call.rs

```diff
@@ -18,59 +19,74 @@ pub fn call(
             write_workspace: Permission::Allow,
             ..
         } => {
-            let results: Vec<Result<Result<ZomeCallResponse, _>, _>> =
+            let results: Vec<Result<ZomeCallResponse, WasmError>> =
                 tokio_helper::block_forever_on(async move {
                     join_all(inputs.into_iter().map(|input| async {
                         let Call {
-                            to_cell,
+                            target,
                             zome_name,
                             fn_name,
                             cap_secret,
                             payload,
-                            provenance,
                         } = input;
-                        let cell_id = to_cell.unwrap_or_else(|| {
-                            call_context
+                        let provenance = call_context.host_context.workspace().source_chain()
+                        .as_ref().expect("Must have source chain to know provenance")
+                        .agent_pubkey()
+                        .clone();
```

Comment:
Truthfully, I'm having a little challenge determining in this (and following) section(s), whether the host_context has the exact thing we need for setting provenance in every case, and I'm trusting you've got the stack of calling contexts right.

Reply:
@artbrock hmmm :thinking: it might not in every case, but also it didn't in develop either :sweat: 

i think what you're pointing to is a followup, because the guest doesn't have any additional context beyond what is here, so the host injecting it isn't (newly) breaking anything either